repo/repo.go

```diff
@@ -140,6 +140,10 @@ func (manager *RepoManager) CheckoutRevision(repoPath string, revision string, f
 	if manager.Config.Checkout.UpdateRepos {
 		git.Git("merge", "--ff-only", "origin/"+trimmedRevision)
 	}
+
+	if manager.Config.Checkout.MergeInBase && trimmedRevision != baseRevision {
```

Comment:
Let's check if the `baseRevision` is set.

```suggestion
	if manager.Config.Checkout.MergeInBase && baseRevision != "" && trimmedRevision != baseRevision {
```